### DISCUSSION TITLE
Secretly Tainted Genes

### DISCUSSION THREAD

**Discussion by Rat-Crusher** [FakeFishGames/Barotrauma#15213] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

When you take a tainted gene and combine it with a pure, untainted gene that is not of the same variety (For instance: Tainted Crawler, and Pure Hammerhead), the output will appear as if it is not tainted. However, the gene is tainted still. Had issues purifying the secretly tainted gene as well afterwards.

### Reproduction steps

1: Acquire untainted gene, and a tainted gene of a different variety.
2: Combine them
3: Attempt to purify by adding pure genes into the gene which appears to be pure (despite it not being pure)

### Bug prevalence

Just once

### Single player or multiplayer?

Single player

### Version

v1.6.19.1 (Unto the Breach Update Hotfix 2)

### Which operating system did you encounter this bug on?

Windows

**Comment by Jasontti** [comment]
Tested a fix for this and found no issues.
Closing discussion as resolved.
